// @mongodb-js/saslprep/dist/code-points-src.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.bidirectional_l = exports.bidirectional_r_al = exports.prohibited_characters = exports.non_ASCII_space_characters = exports.commonly_mapped_to_nothing = exports.unassigned_code_points = void 0;
const util_1 = require("./util");
exports.unassigned_code_points = new Set([
    0x0221,
    ...(0, util_1.range)(0x0234, 0x024f),
    ...(0, util_1.range)(0x02ae, 0x02af),
    ...(0, util_1.range)(0x02ef, 0x02ff),
    ...(0, util_1.range)(0x0350, 0x035f),
    ...(0, util_1.range)(0x0370, 0x0373),
    ...(0, util_1.range)(0x0376, 0x0379),
    ...(0, util_1.range)(0x037b, 0x037d),
    ...(0, util_1.range)(0x037f, 0x0383),
    0x038b,
    0x038d,
    0x03a2,
    0x03cf,
    ...(0, util_1.range)(0x03f7, 0x03ff),
    0x0487,
    0x04cf,
    ...(0, util_1.range)(0x04f6, 0x04f7),
    ...(0, util_1.range)(0x04fa, 0x04ff),
    ...(0, util_1.range)(0x0510, 0x0530),
    ...(0, util_1.range)(0x0557, 0x0558),
    0x0560,
    0x0588,
    ...(0, util_1.range)(0x058b, 0x0590),
    0x05a2,
    0x05ba,
    ...(0, util_1.range)(0x05c5, 0x05cf),
    ...(0, util_1.range)(0x05eb, 0x05ef),
    ...(0, util_1.range)(0x05f5, 0x060b),
    ...(0, util_1.range)(0x060d, 0x061a),
    ...(0, util_1.range)(0x061c, 0x061e),
    0x0620,
    ...(0, util_1.range)(0x063b, 0x063f),
    ...(0, util_1.range)(0x0656, 0x065f),
    ...(0, util_1.range)(0x06ee, 0x06ef),
    0x06ff,
    0x070e,
    ...(0, util_1.range)(0x072d, 0x072f),
    ...(0, util_1.range)(0x074b, 0x077f),
    ...(0, util_1.range)(0x07b2, 0x0900),
    0x0904,
    ...(0, util_1.range)(0x093a, 0x093b),
    ...(0, util_1.range)(0x094e, 0x094f),
    ...(0, util_1.range)(0x0955, 0x0957),
    ...(0, util_1.range)(0x0971, 0x0980),
    0x0984,
    ...(0, util_1.range)(0x098d, 0x098e),
    ...(0, util_1.range)(0x0991, 0x0992),
    0x09a9,
    0x09b1,
    ...(0, util_1.range)(0x09b3, 0x09b5),
    ...(0, util_1.range)(0x09ba, 0x09bb),
    0x09bd,
    ...(0, util_1.range)(0x09c5, 0x09c6),
    ...(0, util_1.range)(0x09c9, 0x09ca),
    ...(0, util_1.range)(0x09ce, 0x09d6),
    ...(0, util_1.range)(0x09d8, 0x09db),
    0x09de,
    ...(0, util_1.range)(0x09e4, 0x09e5),
    ...(0, util_1.range)(0x09fb, 0x0a01),
    ...(0, util_1.range)(0x0a03, 0x0a04),
    ...(0, util_1.range)(0x0a0b, 0x0a0e),
    ...(0, util_1.range)(0x0a11, 0x0a12),
    0x0a29,
    0x0a31,
    0x0a34,
    0x0a37,
    ...(0, util_1.range)(0x0a3a, 0x0a3b),
    0x0a3d,
    ...(0, util_1.range)(0x0a43, 0x0a46),
    ...(0, util_1.range)(0x0a49, 0x0a4a),
    ...(0, util_1.range)(0x0a4e, 0x0a58),
    0x0a5d,
    ...(0, util_1.range)(0x0a5f, 0x0a65),
    ...(0, util_1.range)(0x0a75, 0x0a80),
    0x0a84,
    0x0a8c,
    0x0a8e,
    0x0a92,
    0x0aa9,
    0x0ab1,
    0x0ab4,
    ...(0, util_1.range)(0x0aba, 0x0abb),
    0x0ac6,
    0x0aca,
    ...(0, util_1.range)(0x0ace, 0x0acf),
    ...(0, util_1.range)(0x0ad1, 0x0adf),
    ...(0, util_1.range)(0x0ae1, 0x0ae5),
    ...(0, util_1.range)(0x0af0, 0x0b00),
    0x0b04,
    ...(0, util_1.range)(0x0b0d, 0x0b0e),
    ...(0, util_1.range)(0x0b11, 0x0b12),
    0x0b29,
    0x0b31,
    ...(0, util_1.range)(0x0b34, 0x0b35),
    ...(0, util_1.range)(0x0b3a, 0x0b3b),
    ...(0, util_1.range)(0x0b44, 0x0b46),
    ...(0, util_1.range)(0x0b49, 0x0b4a),
    ...(0, util_1.range)(0x0b4e, 0x0b55),
    ...(0, util_1.range)(0x0b58, 0x0b5b),
    0x0b5e,
    ...(0, util_1.range)(0x0b62, 0x0b65),
    ...(0, util_1.range)(0x0b71, 0x0b81),
    0x0b84,
    ...(0, util_1.range)(0x0b8b, 0x0b8d),
    0x0b91,
    ...(0, util_1.range)(0x0b96, 0x0b98),
    0x0b9b,
    0x0b9d,
    ...(0, util_1.range)(0x0ba0, 0x0ba2),
    ...(0, util_1.range)(0x0ba5, 0x0ba7),
    ...(0, util_1.range)(0x0bab, 0x0bad),
    0x0bb6,
    ...(0, util_1.range)(0x0bba, 0x0bbd),
    ...(0, util_1.range)(0x0bc3, 0x0bc5),
    0x0bc9,
    ...(0, util_1.range)(0x0bce, 0x0bd6),
    ...(0, util_1.range)(0x0bd8, 0x0be6),
    ...(0, util_1.range)(0x0bf3, 0x0c00),
    0x0c04,
    0x0c0d,
    0x0c11,
    0x0c29,
    0x0c34,
    ...(0, util_1.range)(0x0c3a, 0x0c3d),
    0x0c45,
    0x0c49,
    ...(0, util_1.range)(0x0c4e, 0x0c54),
    ...(0, util_1.range)(0x0c57, 0x0c5f),
    ...(0, util_1.range)(0x0c62, 0x0c65),
    ...(0, util_1.range)(0x0c70, 0x0c81),
    0x0c
